source/common/network/listener_impl.cc

```diff
@@ -58,39 +53,50 @@ ListenerImpl::ListenerImpl(Event::DispatcherImpl& dispatcher, Socket& socket, Li
   }
 
   if (bind_to_port) {
-    listener_.reset(
-        evconnlistener_new(&dispatcher.base(), listenCallback, this, 0, -1, socket.fd()));
+    const int ret = ::listen(socket.fd(), 128);
 
-    if (!listener_) {
+    if (ret == -1) {
```

Comment:
add strerror call now? Also, should we use OsSysCalls for listen mocking with the same return pair for int/errno?